src/completion.rs

```diff
@@ -826,31 +824,42 @@ fn walk_package(
                         package_name: get_package_name(package),
                     }));
                 }
-                MonoType::Int => {
-                    push_var_result(&head.k, VarType::Int)
-                }
-                MonoType::Float => {
-                    push_var_result(&head.k, VarType::Float)
-                }
-                MonoType::Bool => {
-                    push_var_result(&head.k, VarType::Bool)
-                }
-                MonoType::Arr(_) => {
-                    push_var_result(&head.k, VarType::Array)
-                }
-                MonoType::Bytes => {
-                    push_var_result(&head.k, VarType::Bytes)
-                }
-                MonoType::Duration => {
-                    push_var_result(&head.k, VarType::Duration)
-                }
-                MonoType::Regexp => {
-                    push_var_result(&head.k, VarType::Regexp)
-                }
-                MonoType::String => {
-                    push_var_result(&head.k, VarType::String)
-                }
-                _ => {}
+                MonoType::Arr(_) => push_var_result(
```

Comment:
Could we add a comment here to ensure that once Eq is implemented, we can un-nest this? Should we ever un-nest it? My preference is to have flat over nested, but I'm not sure if that preference holds up here.